src/app/core/guards/expired-case/expired-case.guard.ts

```diff
@@ -0,0 +1,22 @@
+import { inject } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivateFn, Router } from '@angular/router';
+import { CaseService } from '@services/case/case.service';
+import { map } from 'rxjs';
+
+export const expiredCaseGuard: CanActivateFn = (route: ActivatedRouteSnapshot) => {
+  const caseService = inject(CaseService);
+  const router = inject(Router);
+  const caseId = route.params.caseId;
+
+  return caseService.getCase(caseId).pipe(
+    map((caseDetails) => {
+      if (caseDetails.isDataAnonymised) {
+        // Redirect to expired case screen
+        router.navigate(['/expired-case']);
+        return false;
+      } else {
+        return true;
+      }
```

Comment:
You could drop the else here.

```suggestion
      if (caseDetails.isDataAnonymised) {
        // Redirect to expired case screen
        router.navigate(['/expired-case']);
        return false;
      }
      return true;
```